src/main/java/seedu/address/storage/JsonAdaptedPerson.java

```diff
@@ -31,11 +32,11 @@
      */
     @JsonCreator
     public JsonAdaptedPerson(@JsonProperty("name") String name, @JsonProperty("phone") String phone,
-                             @JsonProperty("email") String email, @JsonProperty("done") boolean isDone) {
+                             @JsonProperty("email") String email, @JsonProperty("done") Boolean isDone) {
         this.name = name;
         this.phone = phone;
         this.email = email;
-        this.isDone = isDone;
+        this.isDone = Objects.requireNonNullElse(isDone, false);
```

Comment:
Just fyi, Case, where isDone is null, has already been covered.  If you notice that in model.person.isdone, even if the value is null, toString() will return false. 

However, this might not be explicitly stated, i think this addition would make things much clearer.

Reply:
I think a more glaring problem is when isDone from json is NOT either false, null or true. Lets say its a string "ok".
This will crash the whole program. 

My recommendation would be to have this JsonProperty Isdone to be a string. Then have a isDoneValid method in both the model.person.IsDone and modeltype() of JsonAdaptedPerson to check whether its true or false. 

There might be other smaller details that i may have missed out here. Can u transfer this issue to me instead?

Reply:
No the problem happens when 

> Just fyi, Case, where isDone is null, has already been covered. If you notice that in model.person.isdone, even if the value is null, toString() will return false.
> 
> However, this might not be explicitly stated, i think this addition would make things much clearer.

No thats not the main problem. The problem is that the program would crash before it even reaches that point. Say for example, the isDone field does not exist in the JSON file. Trying to assign a null to a boolean in the original line 38 will cause the crash. In fact, the requirenonnull in the IsDone class doesnt even work bcos u cant assign null to boolean in the first place. 





> I think a more glaring problem is when isDone from json is NOT either false, null or true. Lets say its a string "ok". This will crash the whole program.
> 
> My recommendation would be to have this JsonProperty Isdone to be a string. Then have a isDoneValid method in both the model.person.IsDone and modeltype() of JsonAdaptedPerson to check whether its true or false.
> 

This one has been handled alr. If there is a type mismatch an empty address book will be created instead. I mean the same could be said for the other fields as well. What if the jsonproperty of name is a boolean? 

I mean alternatively we could just proceed to make it just generate an empty address book if there happens to be a problem with isdone

Reply:
> This one has been handled alr. If there is a type mismatch an empty address book will be created instead. I mean the same could be said for the other fields as well. What if the jsonproperty of email is a boolean?

My bad, you are right that isdone being a string would be better. The error handling i mentioned above was checking the string itself, not an actual type mismatch check. Apparently, if u define the jsonproperty as a string, it will take what ever was in the json file as a string. I tried giving the boolean value true to a name field in the json file and it just stringified it. So yea, knowing this, it would be better to check against a string

Reply:
> Say for example, the isDone field does not exist in the JSON file. 
I think it still works for this case.

Perhaps we can generate an error message like for phone number.
![image](https://user-images.githubusercontent.com/54056493/135711522-fe110838-0e54-4e99-8e66-f9f89ac7fba0.png)